client/src/containers/frontend/PasswordReset.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component, PropTypes } from 'react';
+import { passwordsAPI } from 'api';
+import { Form } from 'components/global';
+import { entityStoreActions, notificationActions, currentUserActions } from 'actions';
+import { browserHistory } from 'react-router';
+import { get } from 'lodash';
+import { select } from 'utils/entityUtils';
+import { connect } from 'react-redux';
+const { request, flush } = entityStoreActions;
+
+class PasswordReset extends Component {
+
+  static propTypes = {};
+
+  constructor() {
+    super();
+    this.state = {
+      password: '',
+      passwordConfirmation: ''
+    };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  static mapStateToProps(state, ownProps) {
+    return {
+      response: get(state.entityStore.responses, 'reset-password'),
+      users: get(state.entityStore.entities, 'users')
+    };
+  }
+
+  componentWillUnmount() {
+    this.props.dispatch(flush(['reset-password']));
+  }
+
+  handleSubmit(event) {
+    event.preventDefault(event.target);
+    this.props.dispatch(request(passwordsAPI.update(this.state.password, this.state.passwordConfirmation, this.props.params.resetToken), 'reset-password')).promise.then((response) => {
+      this.postUpdate(response.data);
```

Comment:
No... the idea I'm getting at is that this isn't really a request that can be handled within the component like this. Take this approach....

1. Make a new action in src/actions/currentuser.js with the type "START_RESET_PASSWORD". Like this:

export const resetPassword = createAction('START_RESET_PASSWORD', resetPasswordHash => resetPasswordHash);

2. In the component, dispatch the action like this:

`dispatch(resetPassword({hash, password, passwordConfirmation});`

3. If you look in the redux panel, you'll see this action get dispatched with the payload needed to reset the PW. Because this is an asynchronous action, we can go ahead and deal with it first in middleware. Look at src/store/middleware/currentUserMiddleware.js. Add something like this to the exported function:

```
    if (action.type === "START_RESET_PASSWORD") {
      startResetPassword(payload, dispatch);
    }
```

Write a function above called startResetPassword. It will will a lot like the `authenticateWithPassword` method, only it will be using a different API (`passwordsAPI`). When that API call promise is resolved, you'll get a new token back from the server (just like logging in). In the middleware, set the cookie, like we do for a regular login. Then, dispatch some more auth related methods:

```
    dispatch(actions.setCurrentUser(response));
    dispatch(actions.setAuthToken(authToken));
    dispatch(actions.loginComplete());
```

The authentication reducer will see these and login the user.

We'll need to think a little bit about what happens when the reset password fails. We probably handle it a lot like a failed login, and store the error in resetPwError in the authentication reducer state. 

Take a stab at this and see if you can make progress on it.